core/epubmaker/epubmaker.rb

```diff
@@ -105,38 +105,133 @@ def firstHTMLEdit(file)
 								 							.gsub(/&nbsp;/,"&#160;")
 								 							.gsub(/src="images\//,"src=\"")
-	filecontents
+	return filecontents
+rescue => logstring
+	return ''
+ensure
+	Mcmlln::Tools.logtoJson(@log_hash, logkey, logstring)
 end
 
 # Adding the cover holder to the html file
-def secondHTMLEdit(var)
+def secondHTMLEdit(var, logkey='')
 	filecontents = var.gsub(/<body data-type="book">/,"<body data-type=\"book\"><figure data-type=\"cover\" id=\"bookcover01\"><img src=\"cover.jpg\"/></figure>")
-	filecontents
+	return filecontents
+rescue => logstring
+	return ''
+ensure
+	Mcmlln::Tools.logtoJson(@log_hash, logkey, logstring)
+end
+
+## wrapping a Mcmlln::Tools method in a new method for this script; to return a result for json_logfile
+def overwriteFile(path, filecontents, logkey='')
+	Mcmlln::Tools.overwriteFile(path, filecontents)
+rescue => logstring
+ensure
+    Mcmlln::Tools.logtoJson(@log_hash, logkey, logstring)
+end
+
+## wrapping Bkmkr::Tools.runnode in a new method for this script; to return a result for json_logfile
+def runNode_epubmaker(jsfile, htmlfile, logkey='')
+	Bkmkr::Tools.runnode(jsfile, htmlfile)
+rescue => logstring
+ensure
+    Mcmlln::Tools.logtoJson(@log_hash, logkey, logstring)
+end
+
+def cdToProjectTmp(logkey='')
+	FileUtils.cd(Bkmkr::Paths.project_tmp_dir)
+rescue => logstring
+ensure
+    Mcmlln::Tools.logtoJson(@log_hash, logkey, logstring)
+end
+
+## wrapping Bkmkr::Tools.processxsl in a new method for this script; to return a result for json_logfile
+def processxsl_epubmaker(epub_tmp_html, epub_xsl, tmp_epub, convert_log_txt, logkey='')
+	Bkmkr::Tools.processxsl(epub_tmp_html, epub_xsl, tmp_epub, convert_log_txt)
+rescue => logstring
+ensure
+    Mcmlln::Tools.logtoJson(@log_hash, logkey, logstring)
 end
 
 # fix cover.html doctype
-def firstCoverEdit(file)
+def firstCoverEdit(file, logkey='')
 	covercontents = File.read(file).gsub(/&lt;!DOCTYPE html&gt;/,"<!DOCTYPE html>").gsub(/<body>/,"<body><h1 class=\"Nonprinting\">Cover</h1>")
-	covercontents
+	return covercontents
+rescue => logstring
+	return ''
+ensure
+	Mcmlln::Tools.logtoJson(@log_hash, logkey, logstring)
 end
 
 # fix cover ncx entry
-def firstNCXEdit(file)
+def firstNCXEdit(file, logkey='')
 	ncxcontents = File.read(file).gsub(/<text\/><\/navLabel><content src="\#bookcover01"\/>/,"<text>Cover</text></navLabel><content src=\"cover.html\"/>")
-	ncxcontents
+	return ncxcontents
+rescue => logstring
+	return ''
+ensure
+	Mcmlln::Tools.logtoJson(@log_hash, logkey, logstring)
 end
 
 # fix author info in opf
-def firstOPFEdit(file)
+def firstOPFEdit(file, logkey='')
 	opfcontents = File.read(file).gsub(/<dc:creator/,"<dc:identifier id='isbn'>#{Metadata.eisbn}</dc:identifier><dc:creator id='creator'")
-	opfcontents
+	return opfcontents
+rescue => logstring
+	return ''
+ensure
+	Mcmlln::Tools.logtoJson(@log_hash, logkey, logstring)
+end
+
+def convertCoverImg(file, logkey='')
+	unless Bkmkr::Tools.processimages == "false"
```

Comment:
Nah, leave it the way you have it. It would only potentially be an issue for non-Macmillan folks, and I'm still undecided about whether covers should have their own handling or be lumped in with other images.